docs/developer/docker.md

```diff
@@ -0,0 +1,25 @@
+# Development and the docker environment
+
+We use docker and dokku in production, and docker-compose can be used as a local development environment.
+
+Start everything up with `docker-compose up`. This will show you logs from all of the running containers in the console. To run it in the background, use `docker-compose up -d`. Shut it down with `docker-compose down`.
+
+To see the logs (eg. in another console, or if you're running it in the background), run `docker-compose logs`. To see logs for a particular container, run `docker-compose logs [containter]`, eg. `docker-compose logs redis` (with the service names from the `docker-compose` file). Use the `docker-compose logs -f` to continually show the logs as the service runs.
+
+## Updating the Dockerfile
+
+If you make changes to either `Dockerfile` or `docker-compose.yml` you'll need to rebuild it locally to test it:
+
+```
+$ docker-compose down # (if running)
+$ docker-compose build --no-cache
+$ docker-compose up # (to restart)
+```
+
+## Updating the code
+
+You'll need to rebuild the docker environment if you [add, remove, or upgrade the dependencies](dependencies.md).
+
+You don't need to rebuild or restart the containers when you edit any Python code. The changes will be detected and the server restarts automatically. 
```

Comment:
> The changes will be detected and the server restarts automatically.

Is that true? I don't think it is. Our gunicorn setup is the same for dev and live. I just tested locally by sticking in a "return 1/0" and I don't see the changes until I stop and docker-up again.

What is true is that you don't have to **build** after only changing python and that is worth noting; but pretty sure you have to restart containers